app/Models/TestOutput.php

```diff
@@ -44,21 +44,53 @@ public function tests(): HasMany
     /**
      * Returns uncompressed test output.
      */
-    public static function DecompressOutput($output)
+    public static function DecompressOutput(mixed $output): string
     {
+        if (is_resource($output)) {
+            $output = stream_get_contents($output);
+        }
+
+        if (!is_string($output)) {
+            $output = '';
+        }
+
         if (!config('cdash.use_compression')) {
             return $output;
         }
-        if (config('database.default') === 'pgsql') {
-            if (is_resource($output)) {
-                $output = base64_decode(stream_get_contents($output));
-            } else {
-                $output = base64_decode($output);
+
+        // This output could be:
+        // - compressed
+        // - compressed and base64 encoded
+        // - base64 encoded but not compressed
+        // - neither base64 encoded nor compressed
+
+        // Check if output is compressed.
+        @$decompressed = gzuncompress($output);
+        if ($decompressed !== false) {
+            return $decompressed;
+        }
```

Comment:
Should we add a comment here indicating that compression always happens before base64 encoding?